client/question/services/FeedbackGeneratorService.js

```diff
@@ -62,6 +95,38 @@ tie.factory('FeedbackGeneratorService', [
     var previousRuntimeErrorString = '';
 
     /**
+     * Pseudo-randomly returns an int between min (inclusive) and max.
+     *
+     * @param {number} min Minimum number for range (inclusive).
+     * @param {number} max Maximum number for range (exclusive).
+     * @returns {number}
+     * @private
+     */
+    var _getRandomInt = function(min, max) {
+      var minimum = Math.ceil(min);
+      var maximum = Math.floor(max);
+      return Math.floor(Math.random() * (maximum - minimum)) + minimum;
+    };
+
+    /**
+     * Randomly selects feedback string from an object containing variations
+     * of a given feedback type.
+     *
+     * @param {string} type Correctness feedback type.
```

Comment:
If this is functionally identical to state then let's just use the terminology "state" throughout (and call this arg correctnessFeedbackState or similar).

Reply:
I don't think they are functionally identical. I also feel using "state" to label feedback text type is actually more confusing.